src/vault/src/strategies/strategy.rs

```diff
@@ -114,6 +115,17 @@ pub trait IStrategy: Send + Sync+  BasicStrategy  {
 
             save_strategy(self.clone_self());
 
+            // Add event for deposit
+            EventService::instance().borrow_mut().add_user_event(
```

Comment:
instance will make code a bit more complex than it can be. Please use new + method, or pure function. Another way is to use some instance service common for all classes